src/server/frontend_wayland/window_wl_surface_role.cpp

```diff
@@ -387,7 +403,10 @@ auto mf::WindowWlSurfaceRole::latest_timestamp() const -> std::chrono::nanosecon
 
 void mf::WindowWlSurfaceRole::commit(WlSurfaceState const& state)
 {
-    surface->commit(state);
+    if (surface)
```

Comment:
You can early-return here, right? There's no point in continuing here if `!surface`, because the surface can never come back?